Assets/MRTK/SDK/Experimental/SpatialAwareness/SurfaceMeshesToPlanes.cs

```diff
@@ -0,0 +1,473 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.SpatialAwareness;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using UnityEngine;
+
+#if PLANE_FINDING_PRESENT
+using Microsoft.MixedReality.Toolkit.SpatialAwareness.Processing;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.SpatialAwareness
+{
+    struct PlaneWithType
```

Comment:
Is there an existing `BaseSpatialAwarenessObject` we could use for this? There's `SpatialAwarenessPlanarObject`, but it doesn't have a type. Should we extend it to `SpatialAwarenessPlanarObjectWithType`? Or just add a type to the existing class, which may or may not be set

Reply:
i will double check. since this is experimental, should it block this PR?

Reply:
I guess since this struct is private, we can leave it as-is. I'm going to push some changes to make the struct a little more strict

Reply:
fixed